src/System.Private.CoreLib/shared/System/Buffers/ArrayPool.cs

```diff
@@ -19,6 +29,11 @@ namespace System.Buffers
     /// </remarks>
     public abstract class ArrayPool<T>
     {
+        // Store the shared arraypool in a field of its derived type so the Jit can "see" it when inlining Shared
+        // and devirtualize its calls.
+        private readonly static TlsOverPerCoreLockedStacksArrayPool<T> s_arrayPool
```

Comment:
@jkotas, at one point you'd expressed concern about typing this field as anything derived from ArrayPool, as you wanted to allow developers to use reflection to set it as a workaround for the pool keeping lots of state around potentially indefinitely.  Are you less concerned about that now that we cull arrays from the pool over time?

Reply:
Correct.